pkg/util/log/logtestutils/log_spy.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package logtestutils
+
+import (
+	"encoding/json"
+	"math"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/log/logpb"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+)
+
+// LogSpy is a simple test utility which intercepts and stores
+// log entries for testing purposes. It is not be used outside
+// of tests.
+type LogSpy struct {
+	testState *testing.T
+
+	filters []func(entry logpb.Entry) bool
+
+	logs []logpb.Entry
+
+	mu struct {
+		syncutil.Mutex
+	}
+}
+
+func NewLogSpy(t *testing.T, filters ...func(entry logpb.Entry) bool) *LogSpy {
+	return &LogSpy{
+		testState: t,
+		filters:   filters,
+		logs:      []logpb.Entry{},
+	}
+}
+
+// Intercept satisfies the log.Interceptor interface.
+// It parses the log entry, checks if it passes the filters if
+// any exist and if it does, stores it in the logs slice.
+func (s *LogSpy) Intercept(body []byte) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	var entry logpb.Entry
+	err := json.Unmarshal(body, &entry)
+
+	if err != nil {
+		s.testState.Fatal(err)
+	}
+
+	for _, filter := range s.filters {
+		if !filter(entry) {
+			return
+		}
+	}
+
+	s.logs = append(s.logs, entry)
+}
+
+// The read functions below discard the logs which are consumed.
```

Comment:
Can update this to conform to go doc comments

Reply:
done.